### PR TITLE
ubuntu-azure-kernel-validation: Refactored the pipeline

### PR DESCRIPTION
**Pull request by adriansuhov**
pipeline-ubuntu-azure-kernel-validation was refactored and now has the
following features:
--Changing testing framework from lisa to lisav2
--Adding Ubuntu 18.10 - available distros now: trusty, xenial, bionic, cosmic
--Adding performance testing on Azure
--Watcher has been changed. It's now looking in apt-cache instead of launchpad website

### CODE REVIEW

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_watcher [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            deleteDir()
+        }
+    }
+}
+
+if (SEND_MAIL == "yes"){
+    node ("meta_slave") {
+        stage('Send-email') {
+          withCredentials([string(credentialsId: 'TEST_EMAIL_LIST', variable: 'MAIL_LIST')]) {
+            echo "Send email to the people that need to confirm the upload."
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_watcher]: wrong message

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_watcher [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            emailext (
+              subject: "Ubuntu-Azure-Pipeline: New kernels detected",
+              to: "${env.MAIL_LIST}",
+              mimeType : "text/html",
+              body: """
+                  Hello,<br/><br/>
+                  New kernels have been detected and testing has been triggered for: ${MESSAGE_LIST}<br/>
+                  You'll be notified again when the automated testing is done.<br/><br/>
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_watcher]: remove this line

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 3]
```diff
@@ -1,211 +1,225 @@
...
 }
 
 properties ([
     parameters([
-        choice (name: 'Distro', choices: 'trusty\nxenial\nbionic\nall', 
-                description: 'trusty - only Trusty validation <br> xenial - only Xenial validation <br> bionic - only Bionic validation <br> all - all distros validation <br>'),
-        choice (name: 'Kernel', choices: 'all\nlinux-azure\nlinux-azure-edge',
-                description: ' all - all kernels validation <br> linux-azure - only linux-azure kernel validation <br> linux-azure-edge - only linux-azure-edge kernel validation - KERNEL NOT SUPPORTED ON AZURE <br>'),
+        choice (name: 'Distro', choices: 'trusty\nxenial\nbionic\ncosmic', 
+                description: 'trusty - Trusty validation <br> xenial - Xenial validation <br> bionic - Bionic validation <br> cosmic - Cosmic validation'),
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: Change this to be like "trusty - 14.04 validation", xenial - 16.04 and so on.

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 4]
```diff
@@ -1,211 +1,225 @@
...
+        choice (name: 'Kernel', choices: 'linux-azure\nlinux-azure-edge',
+                description: 'linux-azure - linux-azure kernel validation <br> linux-azure-edge - linux-azure-edge kernel validation'),
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: Add "latest proposed" wording somewhere in description

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 5]
```diff
@@ -1,211 +1,225 @@
...
         choice (name: 'Platform', choices: 'all\nhyperv\nazure',
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: Azure first

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 6]
```diff
@@ -1,211 +1,225 @@
...
-                description: '<br> all - all platforms validation <br> hyperv - only Hyper-V validation <br> azure - only Azure validation')
+                description: 'all - all platforms validation <br> hyperv - only Hyper-V validation <br> azure - only Azure validation'),
+        choice (name: 'Performance', choices: 'yes\nno',
+                description: 'yes - will do Performance testing <br> no - Performance testing will be skipped'),
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: will do Azure Performance testing

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 7]
```diff
@@ -1,211 +1,225 @@
...
-KERNEL_AZURE = "linux-azure"
-KERNEL_AZURE_EDGE = "linux-azure-edge"
-KERNEL_ALL = "all"
-
+def ARM_DISTRO_HASH = [trusty:'Canonical UbuntuServer 14.04.5-LTS latest',
+                  xenial:'Canonical UbuntuServer 16.04-LTS latest',
+                  bionic:'Canonical UbuntuServer 18.04-LTS latest',
+                  cosmic:'Canonical UbuntuServer 18.10 latest']
+def HYPERV_VHD_HASH = [trusty:'Ubuntu_14.04.5\\Ubuntu14.04.5-server.vhdx',
+                  xenial:'ubuntu_16.04.5\\xenial_16.04.3_x64.vhdx',
```
Comment (chvalean) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: use 16.04.5 vhdx

File: scripts/ubuntu_azure_kernel/ubuntu_azure_kernel_watcher.sh [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    latest_azure=$(sudo apt-cache madison linux-azure | grep ${release}-proposed | awk '{print $3}')
+    latest_edge=$(sudo apt-cache madison linux-azure-edge | grep ${release}-proposed | awk '{print $3}')
+    if [ ! -z $latest_azure ]; then
+        echo "Latest Azure Kernel for $release is $latest_azure"
+        echo "Old Azure kernel for $release : $azure_release"
+        if [ $azure_release != $latest_azure ]; then
+            echo "We have a new kernel. Triggering $release azure kernel testing"
+            #curl -X POST "${JOB_LOCATION}/buildWithParameters?Distro=${release}&Kernel=linux-azure&Platform=all"
+            # Updating the vlaue in latest_versions.sh
```
Comment (chvalean) [inline scripts/ubuntu_azure_kernel/ubuntu_azure_kernel_watcher.sh]: typo, remove commented code anyway

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 9]
```diff
@@ -1,211 +1,256 @@
...
+                                            " -TestPlatform 'Azure'" +
+                                            " -CustomKernel '${KERNEL_TYPE}'" +
+                                            " -ARMImageName '${ARM_DISTRO}'" +
+                                            " -XMLSecretFile '${GLOBAL_LISAV2_SECRETS}'" +
+                                            " -ExitWithZero" +
+                                            " ${test_cmd}"
+                                        )
+                                        junit "report\\*-junit.xml"
+                                        archiveArtifacts "TestResults\\**\\*"
+                                    deleteDir()
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: fix indendation

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_watcher [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+            deleteDir()
+        }
+    }
+}
+
+if (SEND_MAIL == "yes"){
+    node ("meta_slave") {
+        stage('Send-email') {
+            withCredentials([string(credentialsId: 'TEST_EMAIL_LIST', variable: 'MAIL_LIST')]) {
+                echo "Send email to the people that need to confirm the upload."
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_watcher]: I think this message should be changed

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 11]
```diff
@@ -1,211 +1,256 @@
...
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: ditto

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 12]
```diff
@@ -1,211 +1,256 @@
...
+                                            " -TestPlatform 'HyperV'" +
+                                            " -CustomKernel '${KERNEL_TYPE}'" +
+                                            " -OsVHD '${HYPERV_VHD_PATH}'" +
+                                            " -XMLSecretFile '${GLOBAL_LISAV2_SECRETS}'" +
+                                            " -ExitWithZero" +
+                                            " ${test_cmd}"
+                                        )
+                                        junit "report\\*-junit.xml"
+                                        archiveArtifacts "TestResults\\**\\*"
+                                    deleteDir()
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: ditto

File: linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation [hunk 13]
```diff
@@ -1,211 +1,256 @@
...
+                            timeout (30000) {
+                                withCredentials(bindings: [
+                                file(credentialsId:'GLOBAL_LISAV2_SECRETS', variable: 'GLOBAL_LISAV2_SECRETS')]) {
+                                    dir ("${distro}-${env.BUILD_NUMBER}-${env.BRANCH_NAME}") {
+                                        cleanWs()
+                                        git branch: LISAV2_BRANCH, url: LISAV2_REMOTE
+                                        echo "Run-LisaV2.ps1 -TestPlatform 'Azure' -ARMImageName '${ARM_DISTRO}' -RGIdentifier 'ubuntuAz' -TestLocation '${AZURE_LOCATION}' -CustomKernel '${KERNEL_TYPE}' ${test_cmd} -XMLSecretFile '${GLOBAL_LISAV2_SECRETS}'"
+                                        RunPowershellCommand(".\\Run-LisaV2.ps1" +
+                                            " -TestLocation '${AZURE_LOCATION}'" +
+                                            " -RGIdentifier 'ubuntuAz'" +
```
Comment (ader1990) [inline linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation]: add this to a variable, as you use it multiple times

**Pull request merged**